runtime/coreclr-bridge.m

```diff
@@ -152,6 +152,18 @@
 	*mobj_ref = NULL;
 }
 
+void
+xamarin_mono_object_release (MonoReflectionMethod **mobj)
+{
+	xamarin_mono_object_release ((MonoObject **) mobj);
+}
+
+void
+xamarin_mono_object_release (MonoReflectionType **mobj)
+{
+	xamarin_mono_object_release ((MonoObject **) mobj);
+}
+
```

Comment:
what's the value of having overloads if they all call the same function ?

Reply:
Type safety.

Otherwise the calling code would have to do something like:

```c
MonoReflectionMethod *method = ...;
...
xamarin_mono_object_release ((MonoObject **) &method);
```

with the overloads this works:

```c
MonoReflectionMethod *method = ..:;
...
xamarin_mono_object_release (&method);

Blindly casting pointers to other types will end up biting us at some point (it already has during my CoreCLR work).